compiler/lexer.cpp

```diff
@@ -163,9 +164,43 @@ plungefile(char* name, int try_currentpath, int try_includepaths)
             result = plungequalifiedfile(path);
         }
     }
+
+    set_file_defines(inpfname);
+
     return result;
 }
 
+static void
+set_file_defines(char* file)
+{
+    size_t i, len;
+    char* inptr;
+    char newpath[512], newname[512];
+
+    inptr = NULL;
+    len = strlen(file);
+    for (i = len - 1; i < len; i--) {
+        if (file[i] == '/'
+#if defined WIN32 || defined _WIN32
+            || file[i] == '\\'
+#endif
+        ) {
+            inptr = &file[i + 1];
+            break;
+        }
+    }
+
+    if (inptr == NULL) {
+        inptr = file;
+    }
+
+    snprintf(newpath, sizeof(newpath), "\"%s\"", file);
+    snprintf(newname, sizeof(newname), "\"%s\"", inptr);

```

Comment:
Do we need both of these? Seems like just the path would be sufficient. That would simplify this code a fair bit too.

Reply:
This becomes `file.c_str()` and `inptr.c_str()`